spec/views/posts/index.html.erb_spec.rb

```diff
@@ -1,5 +1,50 @@
 require 'rails_helper'
 
-RSpec.describe 'posts/index.html.erb', type: :view do
-  pending "add some examples to (or delete) #{__FILE__}"
+RSpec.describe 'Post Index', type: :system do
+  describe 'post page' do
+    before(:each) do
+      @user = User.create!(
+        name: 'Daniel',
+        bio: 'BCS',
+        photo_link: 'https://drive.google.com/file/d/1wA3g1uPsU7KtTliYiwYhjHJXNM30k2Aq/view?usp=sharing'
+      )
+      @post = @user.posts.create(title: 'Post1', text: 'This is my first post')
+      @comment = Comment.create(text: 'First comment', user_id: @user.id, post_id: @post.id)
+      Like.create(user_id: @user.id, post_id: @post.id)
+
+      visit(user_posts_path(@user.id))
+    end
+
+    it 'shows the users username' do
+      expect(page).to have_content('Daniel')
+    end
+
+    it 'shows number of posts by user' do
+      expect(page).to have_content('number of posts')
+    end
+
+    it 'shows posts title' do
+      expect(page).to have_content('Post1')
+    end
+
+    it 'shows some of the posts body' do
+      expect(page).to have_content 'This is my first post'
+    end
+
+    it 'shows the first comment on a post' do
+      expect(page).to have_content 'First comment'
+    end
+
+    it 'shows how many comments a post has.' do
+      expect(page).to have_content('Comments')
+    end
+
+    it 'shows how many likes a post has.' do
+      expect(page).to have_content('Likes')
+    end
+
+    it 'shows a section for pagination if there are more posts than fit on the view' do
```

Comment:
- Please make sure tests written are passing, see screenshot below ⬇️ 
![Screenshot_138](https://github.com/Daniel-Kigozi/ruby-blog/assets/69638013/ee92a664-90a9-4dea-9599-3f9e011d5a1c)

- Please add the following tests that are missing ⬇️ 

- [ ] Tests that asserts you can see the user's profile picture.
- [ ] Tests that redirects to the post's show page when you click on a post.